cmd/restic/exclude_test.go

```diff
@@ -189,3 +189,110 @@ func TestMultipleIsExcludedByFile(t *testing.T) {
 		}
 	}
 }
+
+func TestParseSizeStr(t *testing.T) {
+	sizeStrTests := []struct {
+		in       string
+		expected int64
+	}{
+		{"1024", 1024},
+		{"1024b", 1024},
+		{"1024B", 1024},
+		{"1k", 1024},
+		{"100k", 102400},
+		{"100K", 102400},
+		{"10M", 10485760},
+		{"100m", 104857600},
+	}
+
+	for _, tt := range sizeStrTests {
+		actual, err := parseSizeStr(tt.in)
+		test.OK(t, err)
+
+		if actual != tt.expected {
+			t.Errorf("parseSizeStr(%s) = %d; expected %d", tt.in, actual, tt.expected)
+		}
+	}
+}
+
+// TestMultipleIsExcludedByFileSize is for testing that multiple instances of
+// the --exclude-if-present parameter (or the shortcut --exclude-caches do not
+// cancel each other out. It was initially written to demonstrate a bug in
+// rejectIfPresent.
+func TestMultipleIsExcludedByFileSize(t *testing.T) {
+	tempDir, cleanup := test.TempDir(t)
+	defer cleanup()
+
+	// Max size of file is set to be 1k
+	maxSizeStr := "1k"
+
+	// Create some files in a temporary directory.
+	// Files in UPPERCASE will be used as exclusion triggers later on.
+	// We will test the inclusion later, so we add the expected value as
+	// a bool.
+	files := []struct {
+		path string
+		size int64
+		incl bool
+	}{
+		{"42", 100, true},
+
+		// everything in foodir except the FOOLARGE tagfile
+		// should not be included.
+		{"foodir/FOOLARGE", 2048, false},
+		{"foodir/foo", 1002, true},
+		{"foodir/foosub/underfoo", 100, true},
+
+		// everything in bardir except the BARLARGE tagfile
+		// should not be included.
+		{"bardir/BARLARGE", 1030, false},
+		{"bardir/bar", 1000, true},
+		{"bardir/barsub/underbar", 500, true},
+
+		// everything in bazdir should be included.
+		{"bazdir/baz", 100, true},
+		{"bazdir/bazsub/underbaz", 200, true},
+	}
+	var errs []error
+	for _, f := range files {
+		// create directories first, then the file
+		p := filepath.Join(tempDir, filepath.FromSlash(f.path))
+		errs = append(errs, os.MkdirAll(filepath.Dir(p), 0700))
+		file, err := os.OpenFile(p, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
This leaks file descriptors, please close the file after creating it. That should also solve the test failure on Windows.